driver-core/src/main/com/mongodb/connection/DefaultAuthenticator.java

```diff
@@ -16,33 +16,95 @@
 
 package com.mongodb.connection;
 
+import com.mongodb.AuthenticationMechanism;
 import com.mongodb.MongoCredential;
+import com.mongodb.MongoSecurityException;
 import com.mongodb.async.SingleResultCallback;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonString;
 
+import static com.mongodb.AuthenticationMechanism.SCRAM_SHA_1;
+import static com.mongodb.AuthenticationMechanism.SCRAM_SHA_256;
 import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.connection.CommandHelper.executeCommand;
+import static com.mongodb.connection.CommandHelper.executeCommandAsync;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
 
 class DefaultAuthenticator extends Authenticator {
+    private static final ServerVersion FOUR_ZERO = new ServerVersion(3, 7);
+    private static final ServerVersion THREE_ZERO = new ServerVersion(3, 0);
+    private static final BsonString DEFAULT_MECHANISM_NAME = new BsonString(SCRAM_SHA_256.getMechanismName());
+    private static final BsonArray DEFAULT_SUPPORTED_MECHANISMS = new BsonArray(singletonList(DEFAULT_MECHANISM_NAME));
+
     DefaultAuthenticator(final MongoCredential credential) {
         super(credential);
         isTrueArgument("unspecified authentication mechanism", credential.getAuthenticationMechanism() == null);
     }
 
     @Override
     void authenticate(final InternalConnection connection, final ConnectionDescription connectionDescription) {
-        createAuthenticator(connectionDescription).authenticate(connection, connectionDescription);
+        Authenticator authenticator;
+        if (connectionDescription.getServerVersion().compareTo(FOUR_ZERO) < 0) {
```

Comment:
I requested a change on the auth spec PR that's going to complicate this.  See https://github.com/mongodb/specifications/pull/284/commits/e2c1406f1742c4f913f7971660208484c5aa9bbe

Reply:
I already spiked if the list contained a single credential authing that way. So will plug that in.

Reply:
Cool